recipes/flann/all/test_package/CMakeLists.txt

```diff
@@ -2,13 +2,13 @@ cmake_minimum_required(VERSION 3.1)
 project(test_package LANGUAGES C)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
 
 find_package(flann REQUIRED CONFIG)
 
-add_executable(${CMAKE_PROJECT_NAME} test_package.c)
-if(FLANN_SHARED)
-  target_link_libraries(${CMAKE_PROJECT_NAME} flann::flann)
+add_executable(${PROJECT_NAME} test_package.c)
+if(TARGET flann::flann_s)
```

Comment:
What about checking for `BUILD_SHARED_LIBS` instead?

Reply:
It's unrelated.

Reply:
BUILD_SHARED_LIBS controls shared/static libraries built internally by a project, not what type of external library is consumed.

Reply:
Yes, I know, but in the context of Conan and `test_package` the profile used to build the library is the same one used to build the `test_package`. Here `shared` is an option that modifies the package-id, so it will always match.

Reply:
Despite that I think it's a bad usage of BUILD_SHARED_LIBS, I'm pretty sure that injection of BUILD_SHARED_LIBS by conan is unrelated to profile, it's a black magic done by CMake helper when there is a shared option in the Conan class. And test package don't have options.

Reply:
No problem.

My only concern about `if(TARGET flann::flann_s)` is that it won't fail if the assignment in the `package_info` is changed, if we swap this clause the `test_package` will work exactly the same:

```
flann_c_lib = "flann" if self.options.shared else "flann_s"
```

Reply:
Ok I see, it's for pure testing purpose.